engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/BpmnEventTypeTest.java

```diff
@@ -322,6 +322,88 @@ void executeInstance() {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .eventSubProcess(
+                      "subprocess",
+                      sp -> sp.startEvent(elementId()).escalation(escalationCode()).endEvent())
+                  .startEvent()
+                  .intermediateThrowEvent("throw", i -> i.escalation(escalationCode()))
+                  .endEvent()
+                  .done();
+            }
+
+            @Override
+            void executeInstance() {
+              super.executeInstance(Collections.emptyMap());
+            }
+          },
+          new BpmnEventTypeScenario(
+              "Escalation Throw Event",
+              BpmnElementType.INTERMEDIATE_THROW_EVENT,
+              BpmnEventType.ESCALATION) {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .eventSubProcess(
+                      "subprocess",
+                      sp -> sp.startEvent("catch", s -> s.escalation(escalationCode()).endEvent()))
+                  .startEvent()
+                  .intermediateThrowEvent(elementId(), e -> e.escalation(escalationCode()))
+                  .endEvent()
+                  .done();
+            }
+
+            @Override
+            void executeInstance() {
+              super.executeInstance(Collections.emptyMap());
+            }
+          },
+          new BpmnEventTypeScenario(
+              "Escalation Catch Event", BpmnElementType.BOUNDARY_EVENT, BpmnEventType.ESCALATION) {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .startEvent()
+                  .subProcess(
+                      "subprocess",
+                      sp ->
+                          sp.embeddedSubProcess()
+                              .startEvent()
+                              .intermediateThrowEvent("throw", e -> e.escalation(escalationCode()))
+                              .endEvent())
+                  .boundaryEvent(elementId(), b -> b.escalation(escalationCode()))
+                  .cancelActivity(false)
+                  .endEvent()
+                  .done();
+            }
+
+            @Override
+            void executeInstance() {
+              super.executeInstance(Collections.emptyMap());
+            }
+          },
+          new BpmnEventTypeScenario(
+              "Escalation End Event", BpmnElementType.END_EVENT, BpmnEventType.ESCALATION) {
+            @Override
+            BpmnModelInstance modelInstance() {
+              return Bpmn.createExecutableProcess(processId())
+                  .eventSubProcess(
+                      "subprocess",
+                      sp -> sp.startEvent("catch", s -> s.escalation(escalationCode()).endEvent()))
```

Comment:
🔧 We can keep the process simple. We don't need to catch the escalation. We have that covered in other tests. All we're interested in is seeing that the intermediate throw event gets the correct event type.